broker-core/src/test/java/io/zeebe/broker/subscription/message/state/MessageStateControllerTest.java

```diff
@@ -0,0 +1,585 @@
+/*
+ * Zeebe Broker Core
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.broker.subscription.message.state;
+
+import static io.zeebe.util.buffer.BufferUtil.wrapString;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.util.collection.Tuple;
+import io.zeebe.util.sched.clock.ActorClock;
+import java.util.ArrayList;
+import java.util.List;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class MessageStateControllerTest {
+
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  private MessageStateController stateController;
+
+  @Before
+  public void setUp() throws Exception {
+    stateController = new MessageStateController();
+    stateController.open(folder.newFolder("rocksdb"), false);
+  }
+
+  @After
+  public void close() {
+    stateController.close();
+  }
+
+  @Test
+  public void shouldStoreMessage() {
+    // given
+    final Message message = new Message("messageName", "correlationKey", "{\"foo\":\"bar\"}", 1234);
+
+    // when
+    stateController.put(message);
+
+    // then no error
```

Comment:
Maybe nothing was stored? Maybe it was badly serialized? We should check that it can be retrieved properly as well.